packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -43,7 +43,13 @@ export interface Metric {
    * The array may also be empty if the metric value was not based on any
    * entries (e.g. a CLS value of 0 given no layout shifts).
    */
-  entries: PerformanceEntry[] | PerformanceEventTiming[];
+  entries: PerformanceEntry[] | PerformanceEventTiming[] | LayoutShift[];
+}
+
+interface LayoutShift extends PerformanceEntry {
+  value: number;
+  sources: LayoutShiftAttribution[];
+  hadRecentInput: boolean;
```

Comment:
We can be more specific with this type (https://developer.mozilla.org/en-US/docs/Web/API/LayoutShift#instance_properties).

Then with `name: 'layout-shift'` we can type narrow below in the for loop instead of casting `entry` as `LayoutShift`. This also means the below code isn't filtering out non-layout shifts from `entries`.

Reply:
How do you type narrow? Is it just adding a check for `name === 'layout-shift'` before casting?

Reply:
You should not need to cast if it narrows properly, that `name` check _should_ be sufficient

Reply:
the `name` check is not sufficient 😭 it gives a type error because `sources` isn't in `PerformanceEntry`

Reply:
Ah ok, you'll want to use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) then